kube-client/src/lib.rs

```diff
@@ -611,4 +611,137 @@ mod test {
+    #[ignore = "needs cluster for ephemeral containers operations"]
+    async fn can_operate_on_ephemeral_containers() -> Result<(), Box<dyn std::error::Error>> {
+        let pod: Pod = serde_json::from_value(serde_json::json!({
+            "apiVersion": "v1",
+            "kind": "Pod",
+            "metadata": {
+                "name": "ephemeral-container-test",
+                "labels": { "app": "kube-rs-test" },
+            },
+            "spec": {
+                "restartPolicy": "Never",
+                "containers": [{
+                  "name": "busybox",
+                  "image": "busybox:1.34.1",
+                  "command": ["sh", "-c", "sleep 2"],
+                }],
+            }
+        }))?;
+
+        let pod_name = pod.name_any();
+        let client = Client::try_default().await?;
+        let pods = Api::<Pod>::default_namespaced(client);
+
+        // Ephemeral containes can only be applied to a running pod, so one must
+        // be created before any operations are tested.
+        match pods.create(&Default::default(), &pod).await {
+            Ok(o) => assert_eq!(pod.name_unchecked(), o.name_unchecked()),
+            Err(crate::Error::Api(ae)) => assert_eq!(ae.code, 409), // if we failed to clean-up
+            Err(e) => return Err(e.into()),                         // any other case if a failure
+        }
+
+        let current_ephemeral_containers = pods
+            .get_ephemeral_containers(&pod.name_any())
+            .await?
+            .spec
+            .unwrap()
+            .ephemeral_containers;
+
+        // We expect no ephemeral containers initially, get_ephemeral_containers should
+        // reflect that.
+        assert_eq!(current_ephemeral_containers, None);
+
+        let mut busybox_eph: EphemeralContainer = serde_json::from_value(json!(
+            {
+                "name": "myephemeralcontainer1",
+                "image": "busybox:1.34.1",
+                "command": ["sh", "-c", "sleep 2"],
+            }
+        ))?;
+
+        // Attempt to replace ephemeral containers.
+
+        let patch: Pod = serde_json::from_value(json!({
+            "metadata": { "name": pod_name },
+            "spec":{ "ephemeralContainers": [ busybox_eph ] }
+        }))?;
+
+        let current_containers = pods
+            .replace_ephemeral_containers(&pod_name, &PostParams::default(), &patch)
+            .await?
+            .spec
+            .unwrap()
+            .ephemeral_containers
+            .expect("could find ephemeral container");
+
+        // Note that we can't compare the whole ephemeral containers object, as some fields
+        // are set by the cluster. We therefore compare the fields specified in the patch.
+        assert_eq!(current_containers.len(), 1);
+        assert_eq!(current_containers[0].name, busybox_eph.name);
+        assert_eq!(current_containers[0].image, busybox_eph.image);
+        assert_eq!(current_containers[0].command, busybox_eph.command);
```

Comment:
you could maybe rely on [PartialEq for EphemeralContainer](https://docs.rs/k8s-openapi/latest/k8s_openapi/api/core/v1/struct.EphemeralContainer.html#impl-PartialEq%3CEphemeralContainer%3E-for-EphemeralContainer) here and do
```suggestion
        assert_eq!(current_containers[0], busybox_eph);
```

Reply:
oh, nm, i see you say some fields get defaulted in the cluster